File: android-app/app/src/main/java/com/se1853_jv/labverse/data/dto/request/UploadPdfRequest.java
package com.se1853_jv.labverse.data.dto.request;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class UploadPdfRequest implements Serializable {
    private String dataUrl; // URL của PDF file (được upload lên storage bởi backend)
    private String description;
    private List<String> keywords;
    private String title;
    private String authors;
    private String journal;
    private Integer publicationYear;
    private String doi; // Optional - backend will auto-generate if null or empty
    private List<String> tags;
}

File: services/paper-service/src/main/kotlin/com/se1853_jv/dto/request/UploadPdfRequest.kt
package com.se1853_jv.dto.request

import com.se1853_jv.config.annotation.CurrentYear
import jakarta.validation.constraints.Pattern

data class UploadPdfRequest(
    @field:Pattern(
        regexp = "^(https?://)?([\\w\\-]+\\.)+[\\w\\-]+(/[\\w\\-.,@?^=%&:/~+#]*)?$",
        message = "URI must follow the format"
    )
    val dataUrl: String,
    val description: String?,
    val keywords: List<String>?,
    val title: String,
    val authors: String,
    val journal: String,

    @CurrentYear
    val publicationYear: Int,

    @field:Pattern(
        regexp = "(?i)^(?:doi:\\s*|https?://(?:dx\\.)?doi\\.org/)?(10\\.\\d{4,9}/[-._;()/:A-Z0-9]+)$",
        message = "DOI must follow the correct format"
    )
    val doi: String,
    val tags: List<String>?,
)
